src/crypto/random.c

```diff
@@ -15,13 +15,24 @@
  */
 
 #include "crypto.h"
+#include "config.h"
 
 #include <stdint.h>
 
-// TODO: This is linux specific
+#ifdef HAVE_GETRANDOM
 #include <sys/random.h>
+#elif defined(HAVE_ARC4RANDOM)
+#include <stdlib.h>
+#endif
+
 
 void crypto_random(uint8_t* dst, size_t len)
 {
+#ifdef HAVE_GETRANDOM
```

Comment:
It would be nicer that have this be `#if defined` because this is followed with `#elif defined`.